RestTestClient.mutate() should not have side effects

UpStreamThomas opened the pull request:
Before this change, `DefaultRestTestClient.mutate()` was reusing the underlying builder all calls.

Building a new builder for each call clones the `RestTestClientBuilder`, protecting from side effects.

trecloux committed:
fix: RestTestClient.mutate() should not have side effects

Before this change, DefaultRestTestClient.mutate() was reusing the
underlying builder all calls.
Building a new builder for each call clones the RestTestClientBuilder,
protecting from side effects.

Signed-off-by: Thomas Recloux <trecloux@purse.eu>

RemiVangrevelynghe reviewed (approved):
Thanks !

rstoyanchev commented:
Thanks for the pull request .This is done in `.build()` currently, but should be done in `.mutate()` instead.

rstoyanchev commented:
Yes, it should be removed from `.build()`, effectively moved to `.mutate()`.

UpStreamThomas committed:
move RestTestClientBuilder clone to mutate()

Signed-off-by: Thomas Recloux <trecloux@purse.eu>

UpStreamThomas commented:
> Yes, it should be removed from `.build()`, effectively moved to `.mutate()`.

I added a commit with clone removed from `build()`.
I also rebased the branch against main.
